Add `affiliation` to `contact` and `contributor`

briri opened the pull request:
This PR adds an `affiliation` to the `contact` and `contributor` elements for issue #26.

For example:
```
"contact": {
  "name": "John Smith",
  "mbox": "john.smith@tuwien.ac.at",
  "contact_id": {
    "identifier": "https://www.tiss.tuwien.ac.at/person/2351952424",
    "type": "other"
  },
  "affiliation": {
    "name": "Reasearch Data Alliance",
    "affiliation_id": {
      "identifier": "https://ror.org/05dc6w374",
      "type": "ror"
    }
  }
}
```

I did not include an update to ontologies section. I don't have much experience with OWL nor the bandwidth at this point to modify it to include the affiliation.

I'm not sure if there is a defined process for generating new versions of the schema (from the Github) standpoint. I added a new v1.2 here in this PR, but am happy to change it to v1.1 or submit this PR to a new branch if it makes sense to set one up to stage changes prior to their official release.

jomtov commented:
As stated at the RDA20 session on maDMPs in Gothenburg 2023-03-21, we from Stockholm University and other Swedish universities strongly support this PR, as there is a administrative bibliometric / scientometric need for affiliation data in order to track and monitor the entire research output, including DMPs, of an institution.

korbinib commented:
This extension would also allow better usage of maDMPs for creation of projects in RDM systems, application for storage quotas, ...

TomMiksa commented:
Hi,

thanks for this comment.

@MarekSuchanek, @peterneish : we should include this in the next release

borsna commented:
`funder_id.type` needs to have `ror` added

borsna commented on examples/JSON/JSON-schema/1.2/maDMP-schema-1.2.json:
```diff
@@ -0,0 +1,1106 @@
+                        "type": {
+                          "$id": "#/properties/dmp/properties/project/items/properties/funding/properties/funder_id/properties/type",
+                          "type": "string",
+                          "enum": [
```
this list should also include `ror`

Zack-83 replied:
```suggestion
                          "enum": [
                            "ror",
```

Zack-83 commented on ontologies/core/dcso.3.0.0.ttl:
```diff
@@ -336,98 +336,98 @@ Validation with ShEx/SHACL""" .
 dcat:byteSize rdf:type owl:DatatypeProperty ;
-              
+
               rdfs:comment "Byte size" .
```
More correct wording:
```suggestion
              rdfs:comment "Size of the file, in bytes" .
```

briri commented:
Thanks for the feedback @Zack-83. 
We have begin. using ROR ids in our DMP DOI records this past year and I had forgotten to come back and update this PR.

I've added `ror` to the `funder_id` enum and also updated the ontology description for `bytesSize`.

MarekSuchanek commented:
resolved in #133
